ocw_data_parser/ocw_data_parser_test.py

```diff
@@ -2,29 +2,189 @@
 Tests for ocw_data_parser
 """
 import json
-import unittest
+import pytest
+import os
+import shutil
+import responses
+import boto3
+from moto import mock_s3
+from tempfile import TemporaryDirectory
+from ocw_data_parser.ocw_data_parser import CustomHTMLParser, OCWParser
+import ocw_data_parser.test_constants as constants
 
-from ocw_data_parser import OCWParser
+"""
+Tests for OCW Parser
+"""
+
+@pytest.fixture(autouse=True, scope="session")
+def s3_bucket():
+    with mock_s3():
+        conn = boto3.client("s3",
+                            aws_access_key_id="testing",
+                            aws_secret_access_key="testing")
+        conn.create_bucket(Bucket="testing")
+        responses.add_passthru("https://")
+        responses.add_passthru("http://")
+        s3 = boto3.resource("s3",
+                            aws_access_key_id="testing",
+                            aws_secret_access_key="testing")
+        s3_bucket = s3.Bucket(name="testing")
+        yield s3_bucket
+
+@pytest.fixture(autouse=True, scope="function")
+def ocw_parser():
+    """
+    Instantiate an OCWParser object and run functions depending on args passed in
+    """
+    with TemporaryDirectory() as destination_dir:
+        parser = OCWParser(course_dir=constants.COURSE_DIR,
+                            destination_dir=destination_dir,
+                            static_prefix=constants.STATIC_PREFIX)
+        yield parser
+
+@pytest.fixture(autouse=True, scope="session")
+def ocw_parser_s3():
```

Comment:
Can we make use of the `ocw_parser` fixture here:

    @pytest.fixture(autouse=True, scope="session")
    def ocw_parser_s3(ocw_parser):
        ocw_parser.setup_s3_uploading(...)
        yield ocw_parser